syntax = "proto3";
package adam_api;

message GlobalParamsRequest
{
  double learning_rate = 1;
  repeated double left_edge = 2;
  repeated double right_edge = 3;
  int32 epochs = 4;
}

message OptimizeRequest 
{
  repeated double params = 1;
}

message OptimizeResponse 
{
  repeated double result = 1;
}

message EmptyResponse {}

service AdamApiService
{
  rpc SetGlobalParams(GlobalParamsRequest) returns (EmptyResponse);
  rpc Optimize(OptimizeRequest) returns (OptimizeResponse);
}
